src/raft.rs

```diff
@@ -111,7 +111,7 @@ pub struct Raft<T: Storage> {
     /// needs it to be included in a snapshot.
     pub pending_request_snapshot: u64,
 
-    prs: Option<ProgressSet>,
+    prs: UnsafeCell<ProgressSet>,
```

Comment:
How about RefCell? `unsafe` sounds scary.

Reply:
`RefCell` is not zero cost. We already check the implement carefully that it can't be mutate at the same time.

Reply:
What about `Cell`?